### ISSUE TITLE
How do these functions be tested?

### ISSUE THREAD

**Issue by MarryCandLi** [SEI-AMS/pycloud#11]
Hello, I recently reviewed two papers about this project: 1Cyber-Foraging for Improving Survivability of Mobile Systems 2.Establishing Trusted Identities in Disconnected Edge Environments. And I have tried Speech Client (3.0.1) and Cloudlet Client (3.4.1), pycloud (3.0.2) to test the function of communication in the same LAN. My question is as follows:
1 Regarding the “Sign out” function, where is the database storing Username and Password located?
2. Regarding the Devices function, I read the paper saying that it can be connected by Usb/Bluetooth ,but I have not tried it successfully. What should I do specifically?
3.About Migrate ServiceVM, I can find other cloudlets in the same LAN, but there are also problems of  Migrating ServiceVM . What should I do specifically?
Looking forward to your reply.

**Comment by sebastian-echeverria** [comment]
Hello,

1. Regarding the sign-in/sign out functionality, we didn' really implement much in that respect. One of our assumptions was that the cloudlet server is physically secured, so we didn't spend much time handling user access to the Cloudlet Manager GUI, since it should only be accessible locally. And btw, that sign-in is disabled by default; you can see how to enable it [here](https://github.com/SEI-TTG/pycloud/wiki/Enabling-Cloudlet-Security#4-cloudlet-manager-authentication). In any case, there is no user management; the only existing user, "admin", is hardcoded into the Manager code, along with the password. More specifically, it is in pycloud/manager/lib/auth.py.

2. The Pairing of Devices functionality is intended to restrict devices that can communicate through an Android app to the Pycloud API running on the server. It is disabled by default; that is, you can pair a device right away, but it won't do anything unless you enable either Wi-Fi-level security or the API-level security (or both). You can enable either, or both; instructions are [here](https://github.com/SEI-TTG/pycloud/wiki/Enabling-Cloudlet-Security), in sections 2 and 3. In any case, in order to pair a device, you have to 1) go to the Paired Devices page, and Set/Reset Deployment; 2) click on Pair New Device; 3) in the new page, choose USB or Bluetooth; and then 4) proceed with the pairing procedure with the "Pair" button, which will be slightly different depending on the option chosen. In both cases, the Cloudlet Client app needs to be installed on the device first. The USB option requires you to connect the phone to the server through an USB cable, assuming proper drivers are installed. When clicking on Pair, you will have to click "ok" on a prompt on the phone to accept the USB pairing connection. If using Bluetooth, the server needs to have a Bluetooth adapter. Then you will have to go to the Cloudlet Client app before starting the pairing procedure, and and go to the pairing window on the app. The switch to make the device discoverable has to be enabled then, and then the Pair procedure can begin. There may be prompts on the phone or server you have to accept during this procedure. I can write a more step-by step guide on how to do this and add it to the wiki later.

3. The ServiceVM migration requires a bit of setup before it works. Most of what you need to do is described in [this wiki page](https://github.com/SEI-TTG/pycloud/wiki/Enabling-VM-Migration). Before migrating a ServiceVM instance, though, the Service itself needs to be installed in both cloudlets.

I know several of this options are not trivial to configure or test, so if you have more specific questions, feel free to ask and we can discuss them.

**Comment by sebastian-echeverria** [comment]
Ok this new [page](https://github.com/SEI-TTG/pycloud/wiki/Pairing-a-Device) describes the Pairing procedure in a bit more detail. I wrote it mostly from memory so do let me know if some steps are missing or button descriptions do not make sense.

**Comment by MarryCandLi** [comment]
Thank you, I will follow the steps to test  them after a while, and then tell you if it is valid.

**Comment by sebastian-echeverria** [comment]
Closing due to no more comments in 4 months.
